packages/core/src/configs/node/jest.setup.js

```diff
@@ -1,23 +1,108 @@
 const path = require('path')
 
-const camelCase = require('lodash/camelCase')
-
 const { getPaths } = require('@redwoodjs/internal')
+const { defineScenario } = require('@redwoodjs/testing/dist/scenario')
+const { db } = require(path.join(getPaths().api.src, 'lib', 'db'))
+const DEFAULT_SCENARIO = 'standard'
+const PRISMA_RESERVED = ['create', 'connect']
 
-const redwoodPaths = getPaths()
+const findNestedModels = (data) => {
+  let models = []
 
-const { db } = require(path.join(redwoodPaths.api.src, 'lib', 'db'))
+  for (const [field, value] of Object.entries(data)) {
+    if (typeof value === 'object') {
+      if (!models.includes(field) && !PRISMA_RESERVED.includes(field)) {
+        models.push(field)
+      }
+      models = models.concat(findNestedModels(value))
+    }
+  }
 
-/**
- * Loop through all db models and clear the tables before every test
- */
-beforeEach(async () => {
-  for (let model of db._dmmf.datamodel.models) {
-    if (model.name === 'RW_DataMigration') continue
+  return models
+}
 
-    await db[camelCase(model.name)].deleteMany()
+const seedScenario = async (scenario) => {
+  if (scenario) {
+    const scenarios = {}
+    for (const [model, namedFixtures] of Object.entries(scenario)) {
+      scenarios[model] = {}
+      for (const [name, data] of Object.entries(namedFixtures)) {
+        scenarios[model][name] = await db[model].create({ data })
+      }
+    }
+    return scenarios
+  } else {
+    return {}
   }
-})
+}
+
+const removeScenario = async (scenario) => {
+  if (scenario) {
+    let models = []
+
+    for (const [model, namedFixtures] of Object.entries(scenario)) {
+      models.push(model)
+      for (const [_name, data] of Object.entries(namedFixtures)) {
+        models = models.concat(findNestedModels(data))
+      }
+    }
+    // get unique model names only
+    models = Array.from(new Set(models))
+
+    for (const model of models) {
+      await db.$queryRaw(`DELETE FROM ${model}`)
+    }
+  }
+}
+
+window.scenario = (...args) => {
+  let scenarioName, testName, testFunc
+
+  if (args.length === 3) {
+    ;[scenarioName, testName, testFunc] = args
+  } else {
+    scenarioName = DEFAULT_SCENARIO
+    ;[testName, testFunc] = args
+  }
+
+  return window.it(testName, async () => {
+    const path = require('path')
+    const testFileDir = path.parse(window.jasmine.testPath)
+    const testFilePath = `${testFileDir.dir}/${
+      testFileDir.name.split('.')[0]
+    }.scenarios`
+    let allScenarios, scenario, result
+
+    try {
+      allScenarios = require(testFilePath)
+    } catch (e) {
+      // no scenario file found, ignore
+    }
```

Comment:
I think we should check if the file exists and then import it, but not hide the exception. I think it might be frustrating to debug things if there's a syntax error in a scenario.

Reply:
This is the case where there's no scenario file at all...maybe output a warning to the console, but not blow up? Otherwise people that create components without the generator, or anyone that upgrades from any previous version, will have errors all over the place because they won't have any scenario files.

Although how do warnings get output in Jest, will that garble up the test reporting?

Reply:
Actually, I should probably just check in the exception if the error is "file not found" and ignore only that, but re-throw if it's any other error!